test/testpushnotifications.cpp

```diff
@@ -0,0 +1,187 @@
+#include <QTest>
+#include <QVector>
+
+#include "account.h"
+#include "common.h"
+#include "pushnotifications.h"
+#include "creds/abstractcredentials.h"
+
+using namespace std::chrono_literals;
+
+class WebSocketStub : public OCC::AbstractWebSocket
```

Comment:
OK, that's why we have the AbstractWebSocket thing... I still think we should stub the server part instead to test a bit less in a vacuum. This way we're sure it'd behave as the real websocket. It's a bit more work now but that'll pay off ten fold to catch regressions later.